applications/nvidia_video_codec/nvc_decode/cpp/nvc_decode.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <getopt.h>
+#include <cstdlib>
+#include <filesystem>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <holoscan/holoscan.hpp>
+#include <holoscan/operators/holoviz/holoviz.hpp>
+
+// Include the NVIDIA Video Codec operators
+#include "nv_video_decoder.hpp"
+#include "nv_video_reader.hpp"
+
+/**
+ * @brief Operator to print streaming statistics
+ *
+ * This operator collects and prints latency statistics from the video codec pipeline.
+ */
+class StatsOp : public holoscan::Operator {
+ public:
+  HOLOSCAN_OPERATOR_FORWARD_ARGS(StatsOp)
+
+  StatsOp() = default;
+
+  void setup(holoscan::OperatorSpec& spec) override { spec.input<holoscan::gxf::Entity>("input"); }
+
+  void compute(holoscan::InputContext& op_input, holoscan::OutputContext& op_output,
+               holoscan::ExecutionContext& context) override {
+    auto input_message = op_input.receive<holoscan::gxf::Entity>("input");
+
+    if (!first_frame_ignored_) {
+      first_frame_ignored_ = true;
+      return;
+    }
+
+    // For now, just track frame count and processing time
+    frame_count_++;
+
+    // Note: Metadata access from Entity is complex in this SDK version
```

Comment:
To confirm, this is referring to Holoscan SDK? Or the Video Codec SDK?

If this is describing an issue with the Holoscan SDK developer experience, I wonder if we could review internally how we could improve?

Stats tracked here seem OK for demo, do you have suggestions on what additional stats you'd like to track from metadata in the future?

Reply:
I have updated this section to use the values from metadata().